.github/workflows/publish-docker-image.yml

```diff
@@ -0,0 +1,65 @@
+name: "Publish Docker Image"
+
+on:
+  push:
+    tags:
+      - "**"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l -e -o pipefail {0}
+   
+    env:
+      IMAGE_NAME: p4
+      REPOSITORY_OWNER: ${{ github.repository_owner }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: "Fetch Tags"
+        # Workaround for https://github.com/actions/checkout/issues/290
+        run: git fetch --tags --force
+
+      - name: "Get release variables"
+        # RELEASE_VERSION assignment must match the command used in autoconf's
+        # configure.ac file. On update, care should be taken to maintain both
+        # sides congruent.
+        run: |
+          echo 'RELEASE_VERSION='$(git describe --always --tags --dirty) >> $GITHUB_ENV
+          echo 'GIT_URL='$(git remote get-url origin) >> $GITHUB_ENV
+          echo 'GIT_COMMIT='$(git log --pretty=format:'%H' -n 1) >> $GITHUB_ENV
+
+      - name: "Login to Docker registry"
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: "Build Docker image"
+        run: |
+          set -x
```

Comment:
Is `set -x` necessary here as there is only a single command?

Reply:
set -x is not necessary. It's just kept to print commands in the logs. Can be dropped if deemed so.